### ISSUE TITLE
target for single (or few topp) tools?

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#6727]
Use case: Developer needs only his tools and dependencies. Currently, he needs to manually copy his tools and guess all its dll dependencies into a separate folder (potentially including the share/).
Is there a simple e.g., make FlashDeconv, make install possible?

**Comment by jpfeuffer** [comment]
```
if (FlashDeconvOnly)
  install(TARGETS targets...
            RUNTIME_DEPENDENCIES
               [[ARCHIVE|LIBRARY|RUNTIME|OBJECTS|FRAMEWORK|BUNDLE|PRIVATE_HEADER|PUBLIC_HEADER|RESOURCE]
                [DESTINATION <dir>]] <- use our variables so they will be in the places the installer expects
            COMPONENT FlashDeconv
            OPTIONAL EXCLUDE_FROM_ALL
  )
  cpack_add_component(FlashDeconv
                      [DISPLAY_NAME name]
                      [DESCRIPTION description]
                      [HIDDEN | REQUIRED | DISABLED ]
                      [GROUP group]
                      [DEPENDS comp1 comp2 ... ]
                      [INSTALL_TYPES type1 type2 ... ]
                      [DOWNLOADED]
                      [ARCHIVE_FILE filename]
                      [PLIST filename])
  set(CPACK_COMPONENTS_ALL FlashDeconv)
endif()
```

`cmake -DPACKAGE_TYPE="xyz" .`
`make dist`


Would be much much cleaner however, if you do not **guess** your dependencies but actually just specify which of the already defined components you need!

**Comment by jpfeuffer** [comment]
What else would FlashDeconv need except for "OpenMS Libraries"? If you add other libraries to CMake, I hope you will remember.

**Comment by jpfeuffer** [comment]
Also. Of course it will break on Windows since we did not switch to CPack controlled NSIS or WIX, yet. You need to write your own NSIS script or make the current one configurable.

**Comment by jpfeuffer** [comment]
If you knew which dependencies you have, you could just do
`cpack_add_component(FlashDeconv ...)` since we already create install targets for all tools and libraries.

And do `set(CPACK_COMPONENTS_ALL FlashDeconv libraries share)`

**Comment by timosachsenberg** [comment]
I think one main use case is just having a e.g. relocatable folder with the executables that can be sent e.g., on windows as a zip file (instead of a full installer). E.g., a simple copy/ similar to make install would also work.
Can CMake figure out what is needed here (e.g., from the OpenMS lib dependencies)?

**Comment by jpfeuffer** [comment]
There is a CPack zip generator.

**Comment by jpfeuffer** [comment]
I don't know what you are hoping for though. You will always need ALL the dependencies from OpenMS lib.

**Comment by timosachsenberg** [comment]
Yeah, this wouldn't be a problem (e.g., I don't want to reduce the dependencies of the OpenMS lib).
Just a simple way to create an archive for e.g. one tool ideally by just specifying.

something like: 
```
cpack_add_component(FlashDeconv ARCHIVE_FILENAME FlashDeconv.zip) 
set(CPACK_COMPONENTS_ALL FlashDeconv libraries share)
```
or more generic for every tool
```
cpack_add_component(${TOOL_NAME} ARCHIVE_FILENAME "${TOOL_NAME}.zip") 
set(CPACK_COMPONENTS_ALL ${TOOL_NAME} libraries share)
```

Applications would be e.g., FlashDeconv and OpenNuXL.

**Comment by jpfeuffer** [comment]
I forgot, you need to add another install for your tool, since the default install command will put it to COMPONENT TOPP.

I don't know why it is called ALL. https://cmake.org/cmake/help/latest/module/CPackComponent.html#variable:CPACK_COMPONENTS_ALL

`cmake --install --component NAME` works as soon as you added an install command. But it will just copy the files to the install directory (and potentially relinks or whatever you have in other install commands). This does NOT package.

**Comment by timosachsenberg** [comment]
@KyowonJeong : principle of how to package single tools

**Comment by KyowonJeong** [comment]
Thanks a lot for the information! I will try with FLASHDeconv soon!

**Comment by KyowonJeong** [comment]
> other libraries to C

I am not sure what else FLASHDeconv needs. Maybe I need to figure out by trial and error..  For FLASHIda on windows I needed only OpenMS.dll and Qt related ones.

**Comment by KyowonJeong** [comment]
@JeeH-K  could you also check for FLASHDeconvQ and FLASHDeconvWizards?
